sbi/inference/posteriors/direct_posterior.py

```diff
@@ -338,20 +340,27 @@ def sample(
             mcmc_method, mcmc_parameters = self._potentially_replace_mcmc_parameters(
                 mcmc_method, mcmc_parameters
             )
+            transform = mcmc_transform(
+                self._prior, **mcmc_parameters
+            )
             samples = self._sample_posterior_mcmc(
                 num_samples=num_samples,
                 potential_fn=potential_fn_provider(
-                    self._prior, self.net, x, mcmc_method
+                    self._prior, self.net, x, mcmc_method, transform
                 ),
                 init_fn=self._build_mcmc_init_fn(
                     self._prior,
-                    potential_fn_provider(self._prior, self.net, x, "slice_np"),
+                    potential_fn_provider(
+                        self._prior, self.net, x, "slice_np", transform
```

Comment:
unrelated, but why do we default to `slice_np` here? shouldn't the `potential_fn` used for `sir` use the `mcmc_method` passed by the user?